swamp/parsers/tests/test_mtz_parser.py

```diff
@@ -2,39 +2,151 @@
+                        MockGemmiMtzColumn("FreeR_flag", "I"),
+                        MockGemmiMtzColumn("F_XDSdataset", "F"),
+                        MockGemmiMtzColumn("SIGF_XDSdataset", "Q"),
+                        MockGemmiMtzColumn("DANO_XDSdataset", "D"),
+                        MockGemmiMtzColumn("SIGDANO_XDSdataset", "Q"),
+                        MockGemmiMtzColumn("F_XDSdataset(+)", "G"),
+                        MockGemmiMtzColumn("SIGF_XDSdataset(+)", "L"),
+                        MockGemmiMtzColumn("F_XDSdataset(-)", "G"),
+                        MockGemmiMtzColumn("SIGF_XDSdataset(-)", "L"),
+                        MockGemmiMtzColumn("ISYM_XDSdataset", "Y"),
+                        MockGemmiMtzColumn("IMEAN_XDSdataset", "J"),
+                        MockGemmiMtzColumn("SIGIMEAN_XDSdataset", "Q"),
+                        MockGemmiMtzColumn("I_XDSdataset(+)", "K"),
+                        MockGemmiMtzColumn("SIGI_XDSdataset(+)", "M"),
+                        MockGemmiMtzColumn("I_XDSdataset(-)", "K"),
+                        MockGemmiMtzColumn("SIGI_XDSdataset(-)", "M")]
+        parser.reflection_file = MockGemmiMtz(mock_columns)
+        parser.error = False
+        parser.parse()
+
+        self.assertTupleEqual(parser.summary, (b'F_XDSdataset', b'SIGF_XDSdataset', b'IMEAN_XDSdataset',
+                                               b'SIGIMEAN_XDSdataset', b'FreeR_flag', b'DANO_XDSdataset',
+                                               b'SIGDANO_XDSdataset', b'F_XDSdataset(+)', b'SIGF_XDSdataset(+)',
+                                               b'I_XDSdataset(+)', b'SIGI_XDSdataset(+)', b'F_XDSdataset(-)',
+                                               b'SIGF_XDSdataset(-)', b'I_XDSdataset(-)', b'SIGI_XDSdataset(-)'))
+        self.assertEqual(parser.fname, '/empty/path/fname.mtz')
+
+    def test_5(self):
+        parser = MockMtzParser('/empty/path/fname.mtz')
+        mock_columns = [MockGemmiMtzColumn("SIGIminus", "M"),
+                        MockGemmiMtzColumn("F", "F"),
+                        MockGemmiMtzColumn("FWT", "F"),
+                        MockGemmiMtzColumn("F_Plus", "G"),
+                        MockGemmiMtzColumn("f(-)", "G"),
+                        MockGemmiMtzColumn("sigF_PLUS", "L"),
+                        MockGemmiMtzColumn("sigi(+)", "M"),
+                        MockGemmiMtzColumn("SIGFP(-)", "L"),
+                        MockGemmiMtzColumn("dummy-label", "E"),
+                        MockGemmiMtzColumn("Free_FLAG", "I"),
+                        MockGemmiMtzColumn("SIGFP", "Q"),
+                        MockGemmiMtzColumn("sigi", "Q"),
+                        MockGemmiMtzColumn("i", "J"),
+                        MockGemmiMtzColumn("Iplus", "K"),
+                        MockGemmiMtzColumn("iMINUS", "K"),
+                        MockGemmiMtzColumn("sigdano", "Q"),
+                        MockGemmiMtzColumn("danoP", "D")]
```

Comment:
You might wish to add a test with the SA_flag and without the Free flag to ensure that it isn't being picked up - Otherwise these look good!